lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -537,7 +541,11 @@ def canonical_package_id(self):
 
     @property
     def is_virtual(self):
-        return self.is_scm or self.is_subdirs
+        return self.is_scm or self.is_subdirs or self.is_ansible
```

Comment:
What does this influence? When I was designing the concept of virtual deps, I was imagining them corresponding to actual “packages” (collections). Adding the platform into the mix seems to taint that assumption, and I'm not sure if it'd cause problems in other places. That rely on this, corresponding to this ephemeral package concept.

Reply:
The virtual deps are collections, not the virtual namespace itself, and similarly, ansible-core is not a collection. Install/download automatically skip these, and it prevents reaching https://github.com/ansible/ansible/blob/devel/lib/ansible/galaxy/dependency_resolution/providers.py#L422-L425 which would blow up for an ansible-core version that can't be interpreted as SemVer. I added a code comment:

```python
        # TODO: make it easier to plumb non-collection types through the dependency resolver.
        # For example, if is_virtual is False, it shouldn't be assumed that the requirement
        # is a collection, has metadata, has a SemVer version, and is installable.
        # It would be safer to handle expected types explicitly, and error otherwise.
```